pinot-perf/src/main/java/com/linkedin/pinot/perf/BenchmarkQueryEngine.java

```diff
@@ -77,34 +110,60 @@ public void startPinot() throws Exception {
       _perfBenchmarkDriver.addSegment(segmentMetadata);
     }
 
-    System.out.println("Waiting for 10s for everything to be loaded");
-    Uninterruptibles.sleepUninterruptibly(10, TimeUnit.SECONDS);
+    ZkClient client = new ZkClient("localhost:2191", 10000, 10000, new ZNRecordSerializer());
+
+    ZNRecord record = client.readData("/PinotPerfTestCluster/EXTERNALVIEW/" + TABLE_NAME);
+    while (true) {
+      System.out.println("record = " + record);
+      Uninterruptibles.sleepUninterruptibly(10, TimeUnit.SECONDS);
+
+      int onlineSegmentCount = 0;
+      for (Map.Entry<String, Map<String, String>> stringMapEntry : record.getMapFields().entrySet()) {
+        for (Map.Entry<String, String> stringStringEntry : stringMapEntry.getValue().entrySet()) {
+          if (stringStringEntry.getValue().equals("ONLINE")) {
+            onlineSegmentCount++;
+            break;
+          }
+        }
+      }
+
+      System.out.println(onlineSegmentCount + " segments online out of " + segments.length);
+
+      if (onlineSegmentCount == segments.length) {
+        break;
+      }
+
+      record = client.readData("/PinotPerfTestCluster/EXTERNALVIEW/" + TABLE_NAME);
+    }
+
+    ranOnce = false;
+
+    System.out.println(
+        _perfBenchmarkDriver.postQuery(QUERY_PATTERNS[queryPattern], optimizationFlags).toString(2)
```

Comment:
Well, you'd replace the list of patterns in `QUERY_PATTERNS` with queries that you'd like to benchmark. Hopefully you get the table name right. :P